moped-editor/src/views/dashboard/UserSavedViewsTable.js

```diff
@@ -0,0 +1,276 @@
+  const handleSaveClick = useCallback(
+    (id) => () => {
+      setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.View } });
+    },
+    [rowModesModel]
+  );
+
+  // handles canceling adding a new row by clicking the X icon
+  const handleCancelClick = useCallback(
+    (id) => () => {
+      setRowModesModel({
+        ...rowModesModel,
+        [id]: { mode: GridRowModes.View, ignoreModifications: true },
+      });
+      const editedRow = rows.find((row) => row.id === id);
+      if (editedRow.isNew) {
+        setRows(rows.filter((row) => row.id !== id));
+      }
+    },
+    [rowModesModel, rows]
+  );
+
+  // open the delete confirmation modal
+  const handleDeleteOpen = useCallback(
+    (id) => () => {
+      setIsDeleteConfirmationOpen(true);
+      setDeleteConfirmationId(id);
+    },
+    []
+  );
+
+  // handles row delete
+  const handleDeleteClick = (id) => () => {
+    deleteUserSavedView({
+      variables: {
+        id,
+      },
+    })
+      .then(() => {
+        refetch();
+      })
+      .then(() => {
+        // remove row from rows in state
+        setRows(rows.filter((row) => row.id !== id));
+        setIsDeleteConfirmationOpen(false);
+        handleSnackbar(true, "Saved view removed", "success");
+      })
+      .catch((error) => {
+        handleSnackbar(true, "Error removing saved view", "error", error);
+      });
+  };
+
+  const processRowUpdate = (updatedRow, originalRow) => {
+    if (updatedRow.description !== originalRow.description) {
+      return (
+        updateUserSavedView({
+          variables: {
+            id: originalRow.id,
+            description: updatedRow.description,
+          },
+        })
+          .then(() => {
+            refetch();
+            handleSnackbar(true, "Saved view updated", "success");
+          })
+          // from the data grid docs:
+          // Please note that the processRowUpdate must return the row object to update the Data Grid internal state.
+          .then(() => updatedRow)
+          .catch((error) => {
+            handleSnackbar(true, "Error updating saved view", "error", error);
+          })
+      );
+    }
+  };
+
+  const dataGridColumns = useColumns({
+    data,
+    rowModesModel,
+    handleDeleteOpen,
+    handleSaveClick,
+    handleCancelClick,
+    handleEditClick,
+  });
+
+  if (loading || !data) return <CircularProgress />;
+
+  return (
+    <ApolloErrorHandler errors={error}>
+      <DataGridPro
+        sx={dataGridProStyleOverrides}
+        columns={dataGridColumns}
+        rows={rows}
+        autoHeight
+        getRowId={(row) => row.id}
+        rowModesModel={rowModesModel}
+        onRowModesModelChange={handleRowModesModelChange}
+        onProcessRowUpdateError={(error) => console.error}
+        editMode="row"
+        processRowUpdate={processRowUpdate}
+        hideFooter
+        disableRowSelectionOnClick
+        localeText={{ noRowsLabel: "No saved queries to display" }}
```

Comment:
Would be good for this to message to be "No saved **views**..." to align with the tab name, etc

Reply:
great catch! @tillyw could you update the language here to views like John suggested?